### DISCUSSION TITLE
Auto curves for case + CPU/GPU. Case curves always take priority

### DISCUSSION THREAD

**Discussion by ivancuric** [Rem0o/FanControl.Releases#784] [Q&A]
I have a CPU/GPU mix sensor, which takes the max value of either. I'd like to keep them in the 40-70° range.

However, I have a Define 7 Compact which doesn't have a lot of airflow and requires high pressure intake fans that get very loud.

I've set both the CPU and case (intake + exhaust) fans to monitor the CPU/GPU mix temperature:
![image](https://user-images.githubusercontent.com/2827047/168490214-e07dddb2-f157-45d8-bbf2-7e024ed67839.png)

However, what happens is that usually the CPU fan is stuck at 40% during usage spikes (~10s loads), and in that time the intake fans take over which can be quite a bit louder than the CPU fan. It's also kinda misses the point, as the CPU fan should be the priority fan for cooling the CPU.

I've also tried monitoring the case temperature and using simple graphs for the case fans, but it takes a while for it to increase the system temperature, and the extra case airflow helps a lot with the CPU/GPU temps. Do you have any ideas what else I could tweak?

Here's the entire config:

![image](https://user-images.githubusercontent.com/2827047/168490109-d6af3593-8eda-4a5f-8592-17e101aef79e.png)

**Comment by ivancuric** [comment]
A follow-up question:

When should you use a mix sensor and when a mix curve? The functionality seems very similar.

**Reply by Rem0o** [reply]
Mix sensor when the sensors have similar temperature ranges ( like different cores of a same CPU ). 
Mix curves when you want to couple different curve logic together, like a target and a graph, or 2 different graphs that target sensors with different value ranges.

**Comment by Rem0o** [comment]
I think you went a bit overboard with your config to start with and got lost along the way.
Here is my recommendation:

For your CPU Fan: create a graph that is specific to CPU Fan, monitor CPU core. Assign that graph to CPU Fan.

For your case fans:
Create 2 linear curves. One that monitors GPU temp, one that monitors CPU temp. Both should be different, as your GPU temp range is different that your CPU temp range. Your cpu might idle at 30, while your GPU idles at 45 and so on.
Create a mix curve, assign the 2 previous linear curves. Assign that mix to all your intake/exhaust.

Start with that. Tweaks the min/max values of everything until you are happy with the result. If I'm missing a use case, then we can start adding more stuff, but that should be plenty enough.

**Reply by ivancuric** [reply]
Hey thanks! Will try that.

Is it possible to create a curve which doesn't control the fan? I'd like to leave the GPU fans controlled by the card as I'm really happy with how it works. 

Sadly my CPU idles at 40, and easily spikes to 70+. It's a 5600x. My GPU and case temps also hover around 40°.
It could be an airflow problem, which is why I wanted to couple the CPU/GPU usage to airflow as the case sensor is really slow to respond.

**Reply by Rem0o** [reply]
I don't see any GPU control card on your screenshot.

**Reply by ivancuric** [reply]
Exactly. I don't have one. As far as I understand, adding one would override the GPUs built in fan control, which I don't want.

I'd just like to ramp up the case airflow once the CPU or GPU get warmer.

**Reply by Rem0o** [reply]
Then you are just fine as is.

**Reply by ivancuric** [reply]
Thing is, I'm not. The case fans are always the last to ramp up, which is the issue. The CPU will sit at >75° blasting the CPU fan at 100%, and the case fans take forever to ramp up.

I'm sticking to static curves for now.
